### PR TITLE
Handle non-finite floats consistently in filters

### PR DESCRIPTION
**Pull request by dureuill**
# Pull Request

## Related issue

Related meilisearch/meilisearch#3000

### User

- Filters using `field = inf`, (or `infinite`, `NaN`) now match the value as a string rather than returning an internal error.
- Filters using `field < inf` (or other comparison operators) now return an invalid_filter error rather than returning an internal error, much like when using `field < aaa`.

### Implementation

- Add new `NonFiniteFloat` error variants to the filter-parser errors
- Add `Token::parse_as_finite_float` that can fail both when the string is not a float and when the float is not finite
- Refactor `Filter::inner_evaluate` to always use `parse_as_finite_float` instead of just `parse`
- Add corresponding tests

## PR checklist
Please check if your PR fulfills the following requirements:
- [x] Does this PR fix an existing issue, or have you listed the changes applied in the PR description (and why they are needed)?
- [x] Have you read the contributing guidelines?
- [x] Have you made sure that the title is accurate and descriptive of the changes?

Thank you so much for contributing to Meilisearch!

### CODE REVIEW

File: milli/src/search/facet/filter.rs [hunk 1]
```diff
@@ -689,4 +698,61 @@ mod tests {
...
+        let filter = Filter::from_str("price = NaN").unwrap().unwrap();
+        let result = filter.evaluate(&rtxn, &index).unwrap();
+        assert!(result.is_empty());
+        let filter = Filter::from_str("price < NaN").unwrap().unwrap();
+        assert!(matches!(
+            filter.evaluate(&rtxn, &index),
+            Err(crate::Error::UserError(crate::error::UserError::InvalidFilter(_)))
+        ));
+
+        let rtxn = index.read_txn().unwrap();
```
Comment (irevoire) [inline milli/src/search/facet/filter.rs:748]:
```suggestion
```

**Review by irevoire — CHANGES_REQUESTED** [review]
Nice, thank you.
@loiclec if it’s ok for you as well you can merge

**Comment by loiclec** [comment]
Yes, seems perfect :)

**Comment by Kerollmops** [comment]
Could you please check the categories of floats that we accept when we index? Do we also only accept `.is_finite()` numbers?

**Pull request merged**